src/test/kotlin/com/msd/robot/application/GameMapServiceTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.msd.robot.application
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.msd.application.ClientException
+import com.msd.application.GameMapPlanetDto
+import com.msd.application.GameMapService
+import com.msd.planet.domain.PlanetType
+import junit.framework.Assert.assertEquals
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.util.UUID.*
+
+class GameMapServiceTest {
+
+    companion object {
+        val mockGameServiceWebClient = MockWebServer()
+        val gameMapService = GameMapService()
+
+        @BeforeAll
+        @JvmStatic
+        internal fun setUp() {
+            mockGameServiceWebClient.start(port = 8080)
+        }
+
+        @AfterAll
+        @JvmStatic
+        internal fun tearDown() {
+            mockGameServiceWebClient.shutdown()
+        }
+    }
+
+    @Test
+    fun `Returns correct GameMapPlanetDto`() {
+        // given
+        val targetPlanetDto = GameMapPlanetDto(randomUUID(), 3, PlanetType.STANDARD, randomUUID())
+
+        mockGameServiceWebClient.enqueue(
+            MockResponse()
+                .setResponseCode(200)
+                .setBody(jacksonObjectMapper().writeValueAsString(targetPlanetDto))
+        )
+
+        // when
+        val responsePlanetDto = gameMapService.retrieveTargetPlanetIfRobotCanReach(randomUUID(), targetPlanetDto.id)
+
+        // then
+        assertEquals(targetPlanetDto.id, responsePlanetDto.id)
+    }
+
+    @Test
+    fun `Throws InvalidMoveException if the GameMap Service returns a 400`() {
```

Comment:
You should probably also add the exception message to your tests.

Reply:
It doesnt check the message so it has no use, other than making it harder to maintain

Reply:
I mean all tests with exceptions especially the ones where a ClientException ist expected to be thrown. If it doesn't have a message then you don't have to test for it,